src/Workspaces/Core/Portable/Execution/ChecksumTreeCollection.cs

```diff
@@ -95,6 +97,44 @@ public ChecksumObject GetChecksumObject(Checksum checksum, CancellationToken can
             throw ExceptionUtilities.UnexpectedValue(checksum);
         }
 
+        public ImmutableDictionary<Checksum, ChecksumObject> GetChecksumObjects(IEnumerable<Checksum> checksums, CancellationToken cancellationToken)
```

Comment:
immutable dictionaries are pretty expensive.  They cause lots of allocs.  Are you sure a simple IReadOnlyDictionary would not be good enough here?  ImmutableDictionary is good when you actually have multithreaded reading/writing.  But it looks like you just create teh dictionary here and return it out.  Does it need to be mutated after that?  If not, an IReadOnlyDictionary would be better.